File: solutions/Python/66.PlusOne.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:

        length: int = len(digits)

        for i in reversed(range(length)):
            digits[i] += 1
            if digits[i] < 10:
                return digits
            digits[i] = 0

        return [1] + digits

File: solutions/JavaScript/66.PlusOne.js
var plusOne = function (digits) {
    const length = digits.length - 1;

    for (let i = length; i >= 0; i--) {
        digits[i] += 1;
        if (digits[i] < 10) return digits;
        digits[i] = 0;
    }

    digits.unshift(1);
    return digits;
};